### PR TITLE
Simplify Button styling customizations

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR simplifies the way that Button component design patterns are customized (extended) in the client, and updates two components (`AnnotationShareControl`, `ShareAnnotationsPanel`) to use the shared `TextInputWithButton` component, which eliminates the need for a custom `.InputButton` design pattern in the client. The styling for the text-inputs-with-button in those two components is slightly visually different (see below), but there are no other user-visible changes in this work.

This PR achieves simplification by taking advantage of the recently-added support for a `classes` prop to the Button components in the shared package. This allows adding _additional_ CSS classes to Buttons, instead of fully obliterating the component classes, as the `className` prop does. This in turn allows styling to override just what it needs to, without having to muck around with (honestly really mind-meltingly-confusing) mixins.

This in turn allows us to continue towards _not_ providing mixins as part of the frontend-shared package's API, which is a mid-term goal. The existence of a custom-button pattern-library page in this repository made this work much easier than it might have been, yay.

### User-visible changes

`ShareAnnotationsPanel`, before:

Screen Shot 2021-09-10 at 8 33 43 AM

`ShareAnnotationsPanel`, after:

Screen Shot 2021-09-10 at 8 32 53 AM

`AnnotationShareControl`, before:

Screen Shot 2021-09-10 at 8 33 49 AM

`AnnotationShareControl`, after:

Screen Shot 2021-09-10 at 8 33 02 AM

_Note:_ In doing this work, I noticed that the `ShareAnnotationsPanel` doesn't put focus on the text-input field when opened (this pre-existed these changes). I will track that separately, as it should be done as part of migrating the panel to a (non-modal) Dialog overall for focus control, and that in turn has a couple of dependencies.

### CODE REVIEW

File: src/sidebar/components/ShareAnnotationsPanel.js [hunk 1]
```diff
@@ -69,19 +75,20 @@ function ShareAnnotationsPanel({ toastMessenger }) {
...
-                />
+                <TextInputWithButton>
+                  <TextInput
+                    aria-label="Use this URL to share these annotations"
+                    type="text"
+                    value={shareURI}
+                    readOnly
+                  />
+                  <IconButton
+                    className="InputButton"
```
Comment (esanzgar) [inline src/sidebar/components/ShareAnnotationsPanel.js:86]:
```suggestion
                    classes="InputButton"
```

**Review by esanzgar — APPROVED** [review]
I included a couple of comments, otherwise 👍🏼 .

File: src/sidebar/components/Annotation/AnnotationShareControl.js [hunk 2]
```diff
@@ -136,22 +138,23 @@ function AnnotationShareControl({
...
+              <TextInputWithButton>
+                <TextInput
+                  aria-label="Use this URL to share this annotation"
+                  type="text"
+                  value={shareUri}
+                  readOnly
+                  inputRef={inputRef}
+                />
+                <IconButton
+                  className="InputButton"
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/Annotation/AnnotationShareControl.js]: Thank you thank you thank you for catching this! Fixed.

File: src/sidebar/components/ShareAnnotationsPanel.js (hunk 1 above)
Comment (lyzadanger) [reply src/sidebar/components/ShareAnnotationsPanel.js]: Fixed!

**Pull request merged**